### PR TITLE
Added wildcard support in Products parameter of UpdateServicesServer resource

### LINKED ISSUE #13: UpdateServicesServer - Products
As an example i'm trying to update all SQL servers using Update Services 
I will need to create a list like this

Microsoft SQL Server 2008 R2 - PowerPivot for Microsoft Excel 2010               
Microsoft SQL Server 2012                                                        
Microsoft SQL Server 2014                                                        
Microsoft SQL Server 2016                                                        
Microsoft SQL Server PowerPivot for Excel                                        
SQL Server 2000                                                                  
SQL Server 2005                                                                  
SQL Server 2008 R2                                                             
SQL Server 2008                                                                  
SQL Server 2012 Product Updates for Setup                                        
SQL Server 2014-2016 Product Updates for Setup                                   
SQL Server Feature Pack                                                          
SQL Server                                                                       

 if($WsusProduct = $AllWsusProducts | Where-Object {$_.Title -eq $Product})

If the above like could be changes to 

 if($WsusProduct = $AllWsusProducts | Where-Object {$_.Title -like $Product})

it will allow me to do just the following keyword "SQL Server"

### LINKED ISSUE #57: Add Examples
Examples should be added to show how to use the resources.

### LINKED ISSUE #58: [UpdateServicesServer]Products test failed
#### Details of the scenario you tried and the problem that is occurring
I have a specific array of Windows Products that I have entered for the "Products" portion in the UpdateServicesServer resource. When I attempt to load the DSC configuration, I get the following error during test/set:

```
Products test failed.
Cannot validate argument on parameter 'ErrorRecord'. The argument is null. Provide a valid value for the argument, and then try running the command again. At line:1 char:21 + $params = $args[0]; Set-TargetResource @params -Verbose + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
#### Verbose logs showing the problem
See above
#### Suggested solution to the issue
Uncertain at this time. I switched to a single product for the configuration resource & it completed successfully. I believe it has something to do with my product list.
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
See the code here: https://github.com/microsoft/WindowsServerUpdateServicesConfig/blob/master/WindowsServerUpdateServicesConfig.ps1#L70-L73

#### The operating system the target node is running

N/A
#### Version and build of PowerShell the target node is running

Should be running PS 5.1
#### Version of the DSC module that was used

'UpdateServicesDsc' Version 1.2.1

### LINKED ISSUE #61: UpdateServicesServer: Products fails for "Windows Admin Center"
#### Details of the scenario you tried and the problem that is occurring
The `UpdateServicesServer` resource fails to configure Products when there are multiple entries returned for a given name (e.g. "Windows Admin Center" is a Product Family as well as Product):
```powershell
$WsusServer = Get-WsusServer
$AllWsusProducts = $WsusServer.GetUpdateCategories()
$ProductCollection = New-Object Microsoft.UpdateServices.Administration.UpdateCategoryCollection
$Product = 'Windows Admin Center'
$WsusProduct = $AllWsusProducts | Where-Object -FilterScript { $_.Title -eq $Product }

# WsusProduct now contains 2 members:
$WsusProduct


Type                      : ProductFamily
ProhibitsSubcategories    : False
ProhibitsUpdates          : True
UpdateSource              : MicrosoftUpdate
UpdateServer              : Microsoft.UpdateServices.Internal.BaseApi.UpdateServer
Id                        : a871ee2d-7499-6719-3781-8c63a1642990
Title                     : Windows Admin Center
Description               : Product Family for Windows Admin Center
ReleaseNotes              : 
DefaultPropertiesLanguage : 
DisplayOrder              : 2147483647
ArrivalDate               : 2/12/2021 4:37:45 PM

Type                      : Product
ProhibitsSubcategories    : True
ProhibitsUpdates          : False
UpdateSource              : MicrosoftUpdate
UpdateServer              : Microsoft.UpdateServices.Internal.BaseApi.UpdateServer
Id                        : b1b8f641-1ff2-4ae6-b247-4fe7503787be
Title                     : Windows Admin Center
Description               : Windows Admin Center
ReleaseNotes              : 
DefaultPropertiesLanguage : 
DisplayOrder              : 2147483647
ArrivalDate               : 2/12/2021 4:37:45 PM
```
This causes the following code to fail:
https://github.com/dsccommunity/UpdateServicesDsc/blob/dd6be4e15d88399dc9152364b0b80ef270f329a0/source/DSCResources/MSFT_UpdateServicesServer/MSFT_UpdateServicesServer.psm1#L589-L592

```powershell
$null = $ProductCollection.Add($WsusServer.GetUpdateCategory($WsusProduct.Id))

Cannot find an overload for "GetUpdateCategory" and the argument count: "1".
At line:1 char:1
+ $null = $ProductCollection.Add($WsusServer.GetUpdateCategory($WsusPro ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodException
    + FullyQualifiedErrorId : MethodCountCouldNotFindBest

```
#### Verbose logs showing the problem
```powershell
VERBOSE: Perform operation 'Invoke CimMethod' with following parameters, ''methodName' = SendConfigurationApply,'className' = MSFT_DSCLocalConfigurationManager,'namespaceName' = root/Mi
crosoft/Windows/DesiredStateConfiguration'.
VERBOSE: An LCM method call arrived from computer SRV1 with user sid S-1-5-21-2574420874-2490326428-1586090461-500.
VERBOSE: [SRV1]: LCM:  [ Start  Set      ]
VERBOSE: [SRV1]: LCM:  [ Start  Resource ]  [[UpdateServicesServer]WSUS]
VERBOSE: [SRV1]: LCM:  [ Start  Test     ]  [[UpdateServicesServer]WSUS]
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS server is Present.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS server configuration.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WUSS server subscription.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS SQL server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server SQL Server is MICROSOFT##WID.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer content directory.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server content directory is E:\.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer update improvement program.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server content update improvement program is False.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server languages.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server languages are System.Collections.Specialized.StringCollection.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server Classifications.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server Classifications are e6cf1350-c01b-414d-a61f-263d14d133b4.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server products.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server products are Windows Server 2016.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer synchronization settings.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server synchronize automatically is False.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server synchronize automatically time of day is 06:33:59.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUSServer number of synchronizations per day is 1.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUSServer client targeting mode is Server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Products test failed.
VERBOSE: [SRV1]: LCM:  [ End    Test     ]  [[UpdateServicesServer]WSUS]  in 1.4690 seconds.
VERBOSE: [SRV1]: LCM:  [ Start  Set      ]  [[UpdateServicesServer]WSUS]
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Configuring Wsus.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Check for previous configuration change.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Configuring WSUS Update Improvement Program.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Configuring WSUS for Microsoft Update.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Configuring WSUS no proxy server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Setting WSUS languages.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Setting WSUS products.
Cannot find an overload for "GetUpdateCategory" and the argument count: "1".
    + CategoryInfo          : NotSpecified: (:) [], CimException
    + FullyQualifiedErrorId : MethodCountCouldNotFindBest
    + PSComputerName        : localhost
 
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Setting WSUS classifications.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Setting WSUS synchronization schedule.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS server is Present.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS server configuration.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WUSS server subscription.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS SQL server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server SQL Server is MICROSOFT##WID.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer content directory.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server content directory is E:\.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer update improvement program.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server content update improvement program is False.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server languages.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server languages are System.Collections.Specialized.StringCollection.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server Classifications.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server Classifications are e6cf1350-c01b-414d-a61f-263d14d133b4.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUS Server products.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server products are *.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Getting WSUSServer synchronization settings.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server synchronize automatically is False.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUS Server synchronize automatically time of day is 06:33:59.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUSServer number of synchronizations per day is 1.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] WSUSServer client targeting mode is Server.
VERBOSE: [SRV1]:                            [[UpdateServicesServer]WSUS] Products test failed.
Cannot validate argument on parameter 'ErrorRecord'. The argument is null. Provide a valid value for the argument, and then try running the command again.
    + CategoryInfo          : InvalidData: (:) [], CimException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,New-InvalidResultException
    + PSComputerName        : localhost
 
VERBOSE: [SRV1]: LCM:  [ End    Set      ]  [[UpdateServicesServer]WSUS]  in 0.6250 seconds.
The PowerShell DSC resource '[UpdateServicesServer]WSUS' with SourceInfo '::7::9::UpdateServicesServer' threw one or more non-terminating errors while running the Set-TargetResource 
functionality. These errors are logged to the ETW channel called Microsoft-Windows-DSC/Operational. Refer to this channel for more details.
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : NonTerminatingErrorFromProvider
    + PSComputerName        : localhost
 
VERBOSE: [SRV1]: LCM:  [ End    Set      ]
The SendConfigurationApply function did not succeed.
    + CategoryInfo          : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 1
    + PSComputerName        : localhost
 
VERBOSE: Operation 'Invoke CimMethod' complete.
VERBOSE: Time taken for configuration job to complete is 2.749 seconds
```
#### Suggested solution to the issue
Adjust the code to process multiple entries in `$WsusProduct` separately:
```powershell
            if ($WsusProduct = $AllWsusProducts | Where-Object -FilterScript { $_.Title -eq $Product })
            {
                $WsusProduct | Foreach-Object -Process {
                    $null = $ProductCollection.Add($_)
                }
            }
```

Similar change would likely be needed as part of solution to #13 

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
Configuration WSUSProduct {
    Import-DscResource -ModuleName 'PSDesiredStateConfiguration'
    Import-DscResource -ModuleName 'UpdateServicesDsc' -ModuleVersion 1.2.1
    
    Node localhost {

        UpdateServicesServer WSUS {
            Ensure = 'Present'
            Languages = 'en'
            Products = 'Windows Admin Center'
        }
    }
}
```

#### The operating system the target node is running
```powershell
OsName               : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
```
#### Version and build of PowerShell the target node is running
```powershell
Name                           Value                                                                                                                                                    
----                           -----                                                                                                                                                    
PSVersion                      5.1.17763.1490                                                                                                                                           
PSEdition                      Desktop                                                                                                                                                  
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                  
BuildVersion                   10.0.17763.1490                                                                                                                                          
CLRVersion                     4.0.30319.42000                                                                                                                                          
WSManStackVersion              3.0                                                                                                                                                      
PSRemotingProtocolVersion      2.3                                                                                                                                                      
SerializationVersion           1.1.0.1                                                                                                                                                  
```

#### Version of the DSC module that was used
```powershell
Name              Version Path                                                                                       
----              ------- ----                                                                                       
UpdateServicesDsc 1.2.1   C:\Program Files\WindowsPowerShell\Modules\UpdateServicesDsc\1.2.1\UpdateServicesDsc.psd1
```

### LINKED ISSUE #66: UpdateServicesServer: Null value passed to New-InvalidResultException
#### Details of the scenario you tried and the problem that is occurring

On line 663 in the `Set-TargetResource` function, the following call is made:

```
New-InvalidResultException -Message $errorMessage -ErrorRecord $_
```

However, unlike previous calls to similar commands (see line 416), this command is not inside a try/catch block. Therefore, the `$_` variable is null and the function errors out without detailing the resource failure. Instead, it leads the user on a wild goose chase tracking down $null values. See #58 for an example.

#### Suggested solution to the issue

Proposals:
1. Replace `$_` with a meaningful value such as "Set-TargetResource has been run. However, the resource is not in the desired state. The following parameters are not set to the desired values: ..."
2. Remove lines 660-664 altogether, as testing is in its own function

I prefer option 2, as the Set function has already performed its work and the state has already changed. The Unit and Integration tests should be providing feedback as to the effectiveness of the Set function, not the Set function itself. If, on subsequent execution of the configuration block, the test continues to fail and `Set-TargetResource` continues to be called, then an issue with the `Set-TargetResource` function should be filed and addressed.

Additionally, Issues #45 and #47 are related to apparent false negatives related to the `Set-TargetResource` calling `Test-TargetResource`

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)

See #58

#### The operating system the target node is running

See #58

#### Version of the DSC module that was used

v1.2.1

### LINKED ISSUE #67: UpdateServicesServer unable to handle duplicate product name values
#### Details of the scenario you tried and the problem that is occurring
Loading new products into an array under "UpdateServicesServer" DSC resource has failed due to WSUS now containing multiple product titles with the same name.
#### Verbose logs showing the problem
```
ERROR - An error occurred running method 'Set' in resource 'UpdateServicesServer' for resourceID '[UpdateServicesServer]UpdateServices': System.Management.Automation.MethodException: Cannot find an overload for "GetUpdateCategory" and the argument count: "1".
```
#### Suggested solution to the issue
Modify this code block to handle product names with the same title [Code Reference](https://github.com/dsccommunity/UpdateServicesDsc/blob/dd6be4e15d88399dc9152364b0b80ef270f329a0/source/DSCResources/MSFT_UpdateServicesServer/MSFT_UpdateServicesServer.psm1#L587-L593)

#### The DSC configuration that is used to reproduce the issue (as detailed as possible) 

Not sharing the DSC configuration, since it's an array of products. But here is the example of multiple products returned from the $AllWsusProducts = $WsusServer.GetUpdateCategories() line that is pulling duplicate product titles.
```powershell
Title                                                                               Description
-----                                                                               -----------
Windows Server 2016                                                                 Windows 10 Server
Windows Server 2016                                                                 Windows Server 2016
Windows Server 2019 and later, Servicing Drivers                                    Windows Server 2019 and later, Servicing Drivers
Windows Server 2019 and later, Upgrade & Servicing Drivers                          Windows Server 2019 and later, Upgrade & Servicing Drivers
Windows Server 2019 Datacenter: Azure Edition Hotpatch                              Windows Server 2019 Datacenter: Azure Edition
Windows Server 2019                                                                 Category for Windows Server 2019
Windows Server 2019                                                                 Windows Server 2019 (RS5 Server) and above
```

#### The operating system the target node is running

```
OsName               : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
```

#### Version and build of PowerShell the target node is running

```
Name                           Value
----                           -----
PSVersion                      5.1.17763.1490
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.1490
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
#### Version of the DSC module that was used

1.2.0

### PR DESCRIPTION
**Pull request by NicolasBn**
#### Pull Request (PR) description

This change adds supports of wildcard in Products parameter of UpdateServicesServer resource.

For example:

```powershell
        UpdateServicesServer 'UpdateServices'
        {
            DependsOn = @(
                '[WindowsFeature]UpdateServices'
            )
            Ensure = 'Present'
            ContentDir = 'C:\WSUS'
            Languages = @('en','fr')
            Products = @(
                'Windows Server*'
            )
            Classifications = @(
                'E6CF1350-C01B-414D-A61F-263D14D133B4', #CriticalUpdates
            )
            SynchronizeAutomatically = $true
            SynchronizeAutomaticallyTimeOfDay = '15:30:00'
            ClientTargetingMode = "Client"
        }
```

This PR cover the problem reported in issues #58 and #66 with `ErrorRecord`.
line 703 of MSFT_UpdateServicesServer.psm1:
```powershell
    New-InvalidResultException -Message $errorMessage
```

In Issue #67 and #61, we have the case of multiple products equals exact product title (like `Windows Admin Center `).
This is fixed in line 623 fo MSFT_UpdateServicesServer.psm1:
```powershell
                foreach ($pdt in $WsusProduct)
```

#### This Pull Request (PR) fixes the following issues

- Fixed #61 
- Fixed #67 
- Fixed #66 
- Fixed #58
- Fixed #57 
- Fixed #13

#### Task list
- [x] Added an entry under the Unreleased section of the change log in the CHANGELOG.md.
      Entry should say what was changed, and how that affects users (if applicable).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [x] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/updateservicesdsc/71)

### CODE REVIEW

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 1]
```diff
@@ -65,16 +66,19 @@ function Get-WsusServer {
...
-                            GUID = '00000000-0000-0000-0000-0000testguid'
-                        }
+                    Name = 'Update Classification'
+                    ID = [pscustomobject]@{
+                        GUID = '00000000-0000-0000-0000-0000testguid'
+                    }
                 }
                 return $UpdateClassification
             }
             $Subscription | Add-Member -MemberType ScriptMethod -Name GetUpdateCategories -Value {
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:84]:
```suggestion

            $Subscription | Add-Member -MemberType ScriptMethod -Name GetUpdateCategories -Value {
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 2]
```diff
@@ -65,16 +66,19 @@ function Get-WsusServer {
...
-                        Name = 'Update Classification'
-                        ID = [pscustomobject]@{
-                            GUID = '00000000-0000-0000-0000-0000testguid'
-                        }
+                    Name = 'Update Classification'
+                    ID = [pscustomobject]@{
+                        GUID = '00000000-0000-0000-0000-0000testguid'
+                    }
                 }
                 return $UpdateClassification
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:81]:
```suggestion

                return $UpdateClassification
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 3]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
     }
 
     $WsusServer | Add-Member -MemberType ScriptMethod -Name GetUpdateClassifications -Value {
         $UpdateClassification = [pscustomobject]@{
+            Name = 'Update Classification'
+            ID = [pscustomobject]@{
+                GUID = '00000000-0000-0000-0000-0000testguid'
+            }
+        }
+        return $UpdateClassification
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:122]:
```suggestion

        return $UpdateClassification
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 4]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
+        }
+        $Subscription | Add-Member -MemberType ScriptMethod -Name StartSynchronization -Value {}
+        $Subscription | Add-Member -MemberType ScriptMethod -Name GetUpdateClassifications -Value {
+            $UpdateClassification = [pscustomobject]@{
                 Name = 'Update Classification'
                 ID = [pscustomobject]@{
                     GUID = '00000000-0000-0000-0000-0000testguid'
                 }
+            }
+            return $UpdateClassification
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:183]:
```suggestion

            return $UpdateClassification
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 5]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
         }
-        return $UpdateClassification
+        $Subscription | Add-Member -Force -MemberType ScriptMethod -Name GetUpdateCategories -Value {
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:186]:
```suggestion

        $Subscription | Add-Member -Force -MemberType ScriptMethod -Name GetUpdateCategories -Value {
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 6]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
+            },
+            [pscustomobject]@{
+                Title = 'Windows Server 2016'
+            },
+            [pscustomobject]@{
+                Title = 'Windows Server 2019'
+            }
+            return $Categories
+        }
+        return $Subscription
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:209]:
```suggestion

        return $Subscription
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 7]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
+            [pscustomobject]@{
+                Title = 'Windows Server 2012'
+            },
+            [pscustomobject]@{
+                Title = 'Windows Server 2016'
+            },
+            [pscustomobject]@{
+                Title = 'Windows Server 2019'
+            }
+            return $Categories
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:206]:
```suggestion

            return $Categories
```

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 [hunk 8]
```diff
@@ -93,30 +97,133 @@ function Get-WsusServer {
...
+        [pscustomobject]@{
+            Title = 'Windows Server 2012'
+        },
+        [pscustomobject]@{
+            Title = 'Windows Server 2016'
+        },
+        [pscustomobject]@{
+            Title = 'Windows Server 2019'
         }
         return $Categories
```
Comment (gaelcolas) [inline Tests/Helpers/ImitateUpdateServicesModule.psm1:239]:
```suggestion

        return $Categories
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -13,10 +13,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated build.ps1 script and build.yaml.
 - Changed default timeout in Wait-Win32ProcessStart function for cab installation.
 
+
 ### Added
 
 - Added TestKitchen files for integration tests
 - Added requiredmodules Sampler.GitHubTasks, powershell-yaml
+- Added wildcard support in Products parameter of UpdatesServicesServer resource.
+ Fix issue #61, #67, #66, #58, #57, #13
```
Comment (gaelcolas) [inline CHANGELOG.md]: This should be in the ### Fixes item.
Best is to have one line per fix (unless duplicate). i.e.
- Fix issue with the thingy #61.
- Fix issue with that other stuff #67.

**Review by gaelcolas — CHANGES_REQUESTED** [review]
Please update those minor changes. Otherwise looks good.

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 1 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:84]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 2 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:81]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 3 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:122]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 4 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:183]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 5 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:186]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 6 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:209]: Fixed

File: Tests/Helpers/ImitateUpdateServicesModule.psm1 (hunk 7 above)
Comment (NicolasBn) [reply Tests/Helpers/ImitateUpdateServicesModule.psm1:206]: Fixed

File: CHANGELOG.md (hunk 9 above)
Comment (NicolasBn) [reply CHANGELOG.md]: I changed the CHANGELOG file. Can you say if it looks good ?

**Review by gaelcolas — APPROVED** [review]
Looks good. Thanks

**Pull request merged**